cmake/checks/check_02_system_features.cmake

```diff
@@ -89,6 +89,12 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Darwin")
   # TODO: MM: Check whether this is still necessary...
   #
   STRIP_FLAG(DEAL_II_LINKER_FLAGS "-rdynamic")
+
+  #
+  # At least on Clang 5.0.0 the template depth is set to 128, which is too low
+  # to compile parts of the library. Fix this by setting a large value.
+  #
+  ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-ftemplate-depth=1024")  
```

Comment:
Why have you put it under "system features" and not just under <code>setup_compiler_flags_gnu.cmake</code>?